catalogue/webapp/pages/images.tsx

```diff
@@ -0,0 +1,283 @@
+import { NextPageContext } from 'next';
+import { useEffect, useState, useContext } from 'react';
+import Router from 'next/router';
+import Head from 'next/head';
+import {
+  CatalogueApiError,
+  CatalogueResultsList,
+  Image,
+} from '@weco/common/model/catalogue';
+import { grid, classNames } from '@weco/common/utils/classnames';
+import convertUrlToString from '@weco/common/utils/convert-url-to-string';
+import CataloguePageLayout from '@weco/common/views/components/CataloguePageLayout/CataloguePageLayout';
+import Paginator from '@weco/common/views/components/Paginator/Paginator';
+import ErrorPage from '@weco/common/views/components/ErrorPage/ErrorPage';
+import {
+  ImagesRouteProps,
+  imagesLink,
+  ImagesRoute,
+  imagesRoutePropsToWorksRouteProps,
+} from '@weco/common/services/catalogue/ts_routes';
+import {
+  ImagesApiProps,
+  imagesRouteToApiUrl,
+} from '@weco/common/services/catalogue/ts_api';
+import Space from '@weco/common/views/components/styled/Space';
+import ImageEndpointSearchResults from '../components/ImageEndpointSearchResults/ImageEndpointSearchResults';
+import SearchForm from '@weco/common/views/components/SearchForm/SearchForm';
+import { getImages } from '../services/catalogue/images';
+import { trackSearch } from '@weco/common/views/components/Tracker/Tracker';
+import useSavedSearchState from '@weco/common/hooks/useSavedSearchState';
+import useHotjar from '@weco/common/hooks/useHotjar';
+import TogglesContext from '@weco/common/views/components/TogglesContext/TogglesContext';
+import SearchTabs from '@weco/common/views/components/SearchTabs/SearchTabs';
+
+type Props = {
+  images: CatalogueResultsList<Image> | CatalogueApiError;
+  imagesRouteProps: ImagesRouteProps;
+  apiProps: ImagesApiProps;
+};
+
+const ImagesPagination = ({
+  query,
+  page,
+  results,
+  imagesRouteProps,
+  setSavedSearchState,
+}) => (
+  <Paginator
+    query={query}
+    currentPage={page || 1}
+    pageSize={results.pageSize}
+    totalResults={results.totalResults}
+    link={imagesLink(
+      {
+        ...imagesRouteProps,
+      },
+      'search/paginator'
+    )}
+    onPageChange={async (event, newPage) => {
+      event.preventDefault();
+      const state = {
+        ...imagesRouteProps,
+        page: newPage,
+      };
+      const link = imagesLink(state, 'search/paginator');
+      setSavedSearchState(state);
+      Router.push(link.href, link.as).then(() => window.scrollTo(0, 0));
+    }}
+  />
+);
+
+const Images = ({ images, imagesRouteProps, apiProps }: Props) => {
+  const [loading, setLoading] = useState(false);
+  const [, setSavedSearchState] = useSavedSearchState(imagesRouteProps);
+  const results: CatalogueResultsList<Image> | CatalogueApiError = images;
+  const { searchPrototype } = useContext(TogglesContext);
+
+  const { query, page } = imagesRouteProps;
+
+  useEffect(() => {
+    trackSearch(apiProps, {
```

Comment:
Could we remove this as I am not sure it's actually in the right format?

Reply:
The reason I ask is that the events pipeline allows any events through as it stands, and I don't want to mess the mapping up as it is already a little messy.